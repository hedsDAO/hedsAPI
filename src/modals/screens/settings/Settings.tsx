import { Modal, ModalBody, ModalCloseButton, ModalContent, ModalHeader, ModalOverlay, Text } from '@chakra-ui/react';
import { BannerForm } from '@/modals/screens/settings/components/BannerForm/BannerForm';
import { ButtonGroup } from '@/modals/screens/settings/components/ButtonGroup/ButtonGroup';
import { DescriptionForm } from '@/modals/screens/settings/components/DescriptionForm/DescriptionForm';
import { ProfilePictureForm } from '@/modals/screens/settings/components/ProfilePictureForm/ProfilePictureForm';
import { useSettings } from '@/modals/screens/settings/components/useSettings/useSettings';
import { SETTINGS_MODAL_TITLE } from '@/modals/screens/settings/models/constants';

/**
 * @function Settings
 * @returns {JSX.Element} The `Settings` component JSX element.
 * @description Displays a modal for editing user settings, such as the user's
 * banner, profile picture, and description. Includes form components for each
 * editable setting and a button group for saving changes.
 */

export const Settings = () => {
  const { isOpen, handleClose } = useSettings();
  return (
    <Modal size="lg" motionPreset="slideInBottom" isCentered isOpen={isOpen} onClose={handleClose}>
      <ModalOverlay />
      <ModalContent m={1}>
        <ModalHeader>
          <Text color="white" fontSize={'sm'} fontFamily={'inter'} letterSpacing="widest">
            {SETTINGS_MODAL_TITLE}
          </Text>
        </ModalHeader>
        <ModalCloseButton color="white" />
        <ModalBody>
          <BannerForm />
          <ProfilePictureForm />
          <DescriptionForm />
          <ButtonGroup />
        </ModalBody>
      </ModalContent>
    </Modal>
  );
};