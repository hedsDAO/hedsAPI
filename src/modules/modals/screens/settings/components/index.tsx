export { default as DescriptionForm } from './DescriptionForm/DescriptionForm';
export { default as ProfilePictureForm } from './ProfilePictureForm/ProfilePictureForm';
export { default as ProfileVisibilityForm } from './ProfileVisibilityForm/ProfileVisibilityForm';
