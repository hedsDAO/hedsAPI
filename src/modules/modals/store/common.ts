/**
 * @name SampleModal
 * @summary types, text and data used for the sample download modal.
 *
 */

export class SampleModalText {
  title = 'Download Sample';
  icon = 'fa-light fa-waveform-lines mr-1 text-xl';
  h1 = 'Requirements';
  req1 = 'bpm:';
  req2 = 'duration:';
  primaryAlertText = 'Public submissions end in:';
  warningAlertText = 'Public submissions have ended.';
  checkboxText = 'I understand the submission requirements.';
  primaryButtonText = 'Download';
  secondaryButtonText = 'Back';
}

/**
 * @name SubmitModal
 * @summary types, text and data used for the sample download modal.
 *
 */

export class SubmitModal {
  title = '';
  icon = '';
  h1 = '';
  req1 = '';
  req2 = '';
  primaryAlertText = '';
  warningAlertText = '';
  checkboxText = '';
  primaryButtonText = '';
  secondaryButtonText = '';
}
