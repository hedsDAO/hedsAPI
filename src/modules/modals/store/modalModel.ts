import { createModel } from '@rematch/core';
import type { RootModel } from '@/models';

export enum Modals {
  SETTINGS_MODAL = 0,
  TWITTER_MODAL,
  NAME_MODAL,
  SAMPLE_MODAL,
  SUBMIT_MODAL,
  MINT_MODAL,
}

export interface ModalState {
  currentModal: Modals;
  nextModal?: Modals;
  isOpen: boolean;
}

export const modalModel = createModel<RootModel>()({
  state: {
    isOpen: false,
  } as ModalState,
  reducers: {
    setModal: (state, currentModal) => ({ ...state, currentModal }),
    setModalOpen: (state, isOpen: boolean) => ({ ...state, isOpen }),
    setNextModal: (state, nextModal: Modals) => ({ ...state, nextModal }),
  },
  effects: () => ({}),
});
