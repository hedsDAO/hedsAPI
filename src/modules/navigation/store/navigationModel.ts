import { createModel } from '@rematch/core';
import type { RootModel } from '@/models';

export interface NavLink {
  name: string;
  path: string;
}

export interface NavigationState {
  isOpen: boolean;
  navLinks: Array<NavLink>;
  brandText: string;
}

export const navigationModel = createModel<RootModel>()({
  state: {
    isOpen: false,
    navLinks: [
      { name: 'Explore', path: '/' },
      { name: 'Tapes', path: '/tapes' },
      { name: 'Artists', path: '/artists' },
      // { name: 'Vote', path: '/vote' },
    ],
    brandText: 'heds',
  } as NavigationState,
  reducers: {
    setIsOpen: (state, isOpen: boolean) => ({ ...state, isOpen }),
  },
  selectors: (slice) => ({
    selectIsOpen() {
      return slice((navigationModel) => navigationModel.isOpen);
    },
    selectAllNavLinks() {
      return slice((navigationModel) => navigationModel.navLinks);
    },
    selectBrandText() {
      return slice((navigationModel) => navigationModel.brandText);
    },
  }),
  effects: () => ({}),
});
