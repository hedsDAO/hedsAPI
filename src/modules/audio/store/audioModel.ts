import { createModel } from '@rematch/core';
import type { RootModel } from '../../../models';
import { TapeData, TrackMetadata, User } from '../../../models/common';

export interface AudioState {
  tapes: Array<TapeData>;
  queue: Array<TrackMetadata>;
  artists: Array<User>;
  currentTrack: number;
  currentTapeId: any;
  currentTape: string;
  isPlaying: boolean;
  isLoading: boolean;
  isShowingQueue: boolean;
  volume: number;
  isSample: boolean;
  currentTime: [string, number];
  duration: [string, number];
}

export const audioModel = createModel<RootModel>()({
  state: {
    isPlaying: false,
    isLoading: false,
    isShowingQueue: false,
    volume: 100,
    queue: [],
    tapes: [],
    artists: [],
  } as AudioState,
  reducers: {
    pushToQueue: (state, [track, tape, artist]: [TrackMetadata, TapeData, User]) => {
      const newState: AudioState = { ...state };
      const queue = [...newState.queue, track];
      const tapes = [...newState.tapes, tape];
      const artists = [...state.artists, artist];
      return { ...state, queue, tapes, artists };
    },
    setIsShowingQueue: (state, isShowingQueue: boolean) => ({...state, isShowingQueue}),
    setIsPlaying: (state, isPlaying: boolean) => ({ ...state, isPlaying }),
    setVolume: (state, volume: number) => ({ ...state, volume }),
    setCurrentTime: (state, currentTime: [string, number]) => ({ ...state, currentTime }),
    setDuration: (state, duration: [string, number]) => ({ ...state, duration }),
    setIsLoading: (state, isLoading: boolean) => ({ ...state, isLoading }),
  },
  effects: () => ({}),
});
