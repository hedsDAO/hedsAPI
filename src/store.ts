import { init, RematchDispatch, RematchRootState } from '@rematch/core';
import loadingPlugin, { ExtraModelsFromLoading } from '@rematch/loading';
import persistPlugin from '@rematch/persist';
import storage from 'redux-persist/lib/storage';
import selectPlugin from '@rematch/select';
import { models, RootModel } from './models';
import { createTransform } from 'redux-persist';
import { AudioModelState } from './hooks/useAudio/models/common';

// Custom transform for isPlaying property
const setIsPlayingFalseTransform = createTransform(
  // Outbound (from state to storage)
  (outboundState: unknown, key) => {
    if (key === 'audioModel') {
      const castedState = outboundState as AudioModelState;
      return {
        ...castedState,
        isPlaying: false,
      };
    }
    return outboundState;
  },
  // Inbound (from storage to state)
  (inboundState: unknown, key) => {
    if (key === 'audioModel') {
      const castedState = inboundState as AudioModelState;
      return {
        ...castedState,
        isPlaying: false,
      };
    }
    return inboundState;
  },
);

type FullModel = ExtraModelsFromLoading<RootModel>;

/** Local Storage Persist for Audio Model */
const persistConfig = {
  key: 'root',
  storage,
  version: 2,
  whitelist: ['audioModel'],
  transforms: [setIsPlayingFalseTransform],
};

export const store = init<RootModel, FullModel>({
  models,
  plugins: [selectPlugin(), loadingPlugin(), persistPlugin(persistConfig)],
  redux: {
    devtoolOptions: {
      disabled: process.env.NODE_ENV !== 'development',
    },
  },
});

export type Store = typeof store;
export type Dispatch = RematchDispatch<RootModel>;
export type RootState = RematchRootState<RootModel, FullModel>;
