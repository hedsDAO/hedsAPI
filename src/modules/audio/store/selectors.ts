import { TrackMetadata } from '@/models/common';
import { RootState } from '@/store';

export const selectActiveTrack = (state: RootState) => state.audioModel?.activeTrack;
export const selectQueue = (state: RootState) => state.audioModel?.queue;
export const selectIsLoading= (state: RootState) => state.audioModel?.isLoading;
export const selectIsShowingPlayer = (state: RootState) => state.audioModel?.isShowingPlayer;
export const selectIsShowingQueue = (state: RootState) => state.audioModel?.isShowingQueue;
export const selectIsQueueEmpty = (state: RootState) => state.audioModel?.queue?.length ? false : true;
export const selectAudioVolume = (state: RootState) => state.audioModel?.volume;
export const selectIsTrackPlaying = (state: RootState) => state.audioModel?.isPlaying;
export const selectTimerSeconds = (state: RootState) => state.audioModel?.timerSeconds;
export const selectCountPlayThreshold = (state: RootState) => state.audioModel?.countPlayThreshold;
export const selectActiveTrackStats = (state: RootState) => state.audioModel?.activeTrack?.stats;