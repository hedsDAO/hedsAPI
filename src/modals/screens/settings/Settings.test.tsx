import { BrowserRouter as Router } from 'react-router-dom';
import { Settings } from '@/modals/screens/settings/Settings';
import { store } from '@/store';
import { renderWithRematchStore } from '@/tests/renderWithRematchStore';

describe('Settings unit', () => {
  beforeAll(async () => {});
  describe('snapshot', () => {
    test('snapshot', async () => {
      const snapshot = renderWithRematchStore(
        <Router>
          <Settings />
        </Router>,
        store,
      );
      expect(snapshot).toMatchSnapshot();
    });
  });
});
