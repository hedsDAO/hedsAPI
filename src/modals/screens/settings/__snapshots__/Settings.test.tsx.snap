// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings unit snapshot snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      aria-hidden="true"
      data-aria-hidden="true"
    />
    <div
      class="chakra-portal"
    >
      <div
        aria-hidden="true"
        class="chakra-modal__overlay css-13d1tkw"
        data-aria-hidden="true"
        style="opacity: 0;"
      />
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
      <div
        data-focus-lock-disabled="false"
      >
        <div
          class="chakra-modal__content-container css-1oxhx2p"
          tabindex="-1"
        >
          <section
            aria-describedby="chakra-modal--body-:r1:"
            aria-labelledby="chakra-modal--header-:r1:"
            aria-modal="true"
            class="chakra-modal__content css-1qg5b5z"
            id="chakra-modal-:r1:"
            role="dialog"
            style="opacity: 0; transform: translateX(0px) translateY(16px) translateZ(0);"
            tabindex="-1"
          >
            <header
              class="chakra-modal__header css-72fd9l"
              id="chakra-modal--header-:r1:"
            >
              <p
                class="chakra-text css-1vqwm2t"
              >
                SETTINGS
              </p>
            </header>
            <button
              aria-label="Close"
              class="chakra-modal__close-btn css-ueeams"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="chakra-icon css-onkibi"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M.439,21.44a1.5,1.5,0,0,0,2.122,2.121L11.823,14.3a.25.25,0,0,1,.354,0l9.262,9.263a1.5,1.5,0,1,0,2.122-2.121L14.3,12.177a.25.25,0,0,1,0-.354l9.263-9.262A1.5,1.5,0,0,0,21.439.44L12.177,9.7a.25.25,0,0,1-.354,0L2.561.44A1.5,1.5,0,0,0,.439,2.561L9.7,11.823a.25.25,0,0,1,0,.354Z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <div
              class="chakra-modal__body css-0"
              id="chakra-modal--body-:r1:"
            >
              <div
                class="css-2fr1h1"
              >
                <input
                  accept="image/*"
                  class="chakra-input css-0"
                  data-testid="banner-input"
                  hidden=""
                  type="file"
                />
                <img
                  class="chakra-image css-1io7w98"
                  data-testid="banner-preview"
                />
                <div
                  class="css-gxjp01"
                  role="button"
                >
                  <div
                    class="chakra-stack css-igigla"
                  >
                    <p
                      class="chakra-text css-6018si"
                    >
                      CHANGE BANNER
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="css-ppihqu"
              >
                <input
                  accept="image/*"
                  class="chakra-input hidden css-0"
                  data-testid="profile-picture-input"
                  type="file"
                />
                <img
                  alt="profile-picture"
                  class="chakra-image css-1r9uduk"
                />
                <div
                  class="css-tps9k3"
                  role="button"
                >
                  <div
                    class="chakra-stack css-igigla"
                  >
                    <p
                      class="chakra-text css-ui0dlw"
                    >
                      CHANGE PROFILE
                    </p>
                  </div>
                </div>
              </div>
              <div
                class="chakra-stack css-pb6zhq"
              >
                <div
                  class="chakra-form-control css-0"
                  role="group"
                >
                  <label
                    class="chakra-form__label css-1lzs8oo"
                    for="field-:r2:"
                    id="field-:r2:-label"
                  >
                    Description
                  </label>
                  <div
                    class="chakra-form__helper-text css-9bha6i"
                    id="field-:r2:-helptext"
                  >
                    Update your profile description.
                  </div>
                  <textarea
                    aria-describedby="field-:r2:-helptext"
                    class="chakra-textarea css-1d29m1r"
                    data-testid="description-input"
                    id="field-:r2:"
                    maxlength="130"
                    placeholder=""
                    rows="3"
                  />
                </div>
              </div>
              <div
                class="css-ihd2hm"
              >
                <div
                  class="css-k008qs"
                >
                  <div
                    class="chakra-fade"
                    style="opacity: 0;"
                  >
                    <div
                      class="css-0"
                    >
                      <p
                        class="chakra-text css-ak6yrd"
                        data-testid="desc-count-error"
                      />
                    </div>
                  </div>
                  <div
                    class="chakra-fade"
                    style="opacity: 0;"
                  >
                    <div
                      class="css-k008qs"
                    >
                      <p
                        class="chakra-text css-1mlwv6f"
                      >
                         
                         
                      </p>
                      <p
                        class="chakra-text css-9iemu3"
                      >
                         / 
                      </p>
                      <p
                        class="chakra-text css-9iemu3"
                      >
                         
                        130
                         
                      </p>
                    </div>
                  </div>
                </div>
                <div
                  class="css-nn803e"
                >
                  <button
                    class="chakra-button css-1smjvvq"
                    data-testid="reset-button"
                    disabled=""
                    type="button"
                  >
                    SAVE
                  </button>
                  <button
                    class="chakra-button css-xruyz4"
                    data-testid="clear-button"
                    disabled=""
                    type="button"
                  >
                    <i
                      class="fa-regular fa-arrow-rotate-right"
                    />
                  </button>
                  <button
                    class="chakra-button css-1s7a4ca"
                    type="button"
                  >
                    <i
                      class="fa-solid fa-arrow-right-from-bracket"
                    />
                  </button>
                </div>
              </div>
            </div>
          </section>
        </div>
      </div>
      <div
        aria-hidden="true"
        data-aria-hidden="true"
        data-focus-guard="true"
        style="width: 1px; height: 0px; padding: 0px; overflow: hidden; position: fixed; top: 1px; left: 1px;"
        tabindex="0"
      />
    </div>
  </body>,
  "container": <div
    aria-hidden="true"
    data-aria-hidden="true"
  />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
